lib/mongoid/criteria.rb

```diff
@@ -55,6 +57,40 @@ def ==(other)
       entries == other
     end
 
+    # Call either Findable#find or Enumerable#find based on the given arguments
+    #
+    # If the arguments match Enumerable#find signature, that will be invoked.
+    # Otherwise Findable#find will be invoked
+    #
+    # The signature of Enumerable#find:
+    #
+    #   find(ifnone = nil) { |obj| block } → obj or nil
+    #   find(ifnone = nil) → an_enumerator
```

Comment:
We do not need to duplicate standard library documentation in our docstrings. Linking to stdlib docs I would say is sufficient and this way our documentation cannot get out of date.

Reply:
Good call - fixed at b58cf1de3